test/topology_custom/test_mv_delete_partitions.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+from cassandra.cqltypes import Int32Type
+
+logger = logging.getLogger(__name__)
+
+async def insert_with_concurrency(cql, values, concurrency):
+    def serialize_int(i):
+        return Int32Type.serialize(i, cql.cluster.protocol_version)
+    def get_replicas(key):
+        return cql.cluster.metadata.get_replicas("ks", serialize_int(key))
+    local_node = get_replicas(0)[0]
+    logger.info(f"Starting writes with concurrency {concurrency}")
+    async def do_inserts(i: int):
+        # Insert all rows with keys with the same modulo as i
```

Comment:
This function is hard to understand, I don't think it works the way this comment claims: There's a while() loop below which may skip some of the i's with the same modulo concurrency as the original i (because they end up on this replica).

Moreover, while some i's are skipped, it seems you go above the limit "values" (i goes above values while i_steps remains below). All of this seems deliberate: It seems the overall intention is just to fill "values" different values - it does NOT matter that they are 0..values-1, so some below "values" are skipped while adding more at the end. If so, this is fine and correct, but just hard to understand.

Reply:
> Moreover, while some i's are skipped, it seems you go above the limit "values" (i goes above values while i_steps remains below). All of this seems deliberate: It seems the overall intention is just to fill "values" different values - it does NOT matter that they are 0..values-1, so some below "values" are skipped while adding more at the end. If so, this is fine and correct, but just hard to understand.

That was my intention but I agree that it's hard to read - I changed the comments and made variable names more self-explanatory, hopefully it's more readable now